logback-classic/src/main/java/ch/qos/logback/classic/spi/ThrowableProxy.java

```diff
@@ -46,39 +48,77 @@
     }
 
     private static final ThrowableProxy[] NO_SUPPRESSED = new ThrowableProxy[0];
+    
+    private static final StackTraceElementProxy[] NO_STACKTRACE = new StackTraceElementProxy[0];
 
     public ThrowableProxy(Throwable throwable) {
+        this(throwable, null, false);
+    }
+    
+    private ThrowableProxy(Throwable throwable, IdentityHashMap<Throwable, Boolean> all, boolean circular) {
```

Comment:
Would you prefer I move the map creation to the public constructor? It would eliminate the lazy construction in the private constructor but would require the creation of the map even if the throwable has no cause or suppressed exceptions.